Modix.Services/Core/AuthorizationService.cs

```diff
@@ -158,20 +161,12 @@ public interface IAuthorizationService
         /// <returns>A <see cref="Task"/> that will complete when the operation has completed.</returns>
         Task OnAuthenticatedAsync(ISelfUser self);
 
-        /// <summary>
-        /// Requires that there be an authenticated guild for the current request.
-        /// </summary>
-        void RequireAuthenticatedGuild();
-
-        /// <summary>
-        /// Requires that there be an authenticated user for the current request.
-        /// </summary>
-        void RequireAuthenticatedUser();
-
         /// <summary>
         /// Requires that the given set of claims be present, for the current request.
         /// </summary>
-        /// <param name="claims">A set of claims to be checked against <see cref="CurrentClaims"/>.</param>
+        /// <param name="requiredClaims">A set of claims to be checked against <see cref="CurrentClaims"/>.</param>
+        ServiceResult CheckClaims(params AuthorizationClaim[] requiredClaims);
+
```

Comment:
Could throw an [Obsolete] on this if we're worried about people accidentally using it in new APIs

Reply:
Not sure why it formatted this comment the way it did, but it was meant to go with `RequireClaims`

Reply:
Good point - added